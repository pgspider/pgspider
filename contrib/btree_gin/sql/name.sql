set enable_seqscan=off;

CREATE TABLE test_name (
	i name
);

INSERT INTO test_name VALUES ('a'),('ab'),('abc'),('abb'),('axy'),('xyz');

CREATE INDEX idx_name ON test_name USING gin (i);

SELECT * FROM test_name WHERE i<'abc' ORDER BY i;
SELECT * FROM test_name WHERE i<='abc' ORDER BY i;
SELECT * FROM test_name WHERE i='abc' ORDER BY i;
SELECT * FROM test_name WHERE i>='abc' ORDER BY i;
SELECT * FROM test_name WHERE i>'abc' ORDER BY i;

EXPLAIN (COSTS OFF) SELECT * FROM test_name WHERE i<'abc' ORDER BY i;
EXPLAIN (COSTS OFF) SELECT * FROM test_name WHERE i<='abc' ORDER BY i;
EXPLAIN (COSTS OFF) SELECT * FROM test_name WHERE i='abc' ORDER BY i;
EXPLAIN (COSTS OFF) SELECT * FROM test_name WHERE i>='abc' ORDER BY i;
EXPLAIN (COSTS OFF) SELECT * FROM test_name WHERE i>'abc' ORDER BY i;
