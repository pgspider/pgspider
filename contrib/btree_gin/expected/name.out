set enable_seqscan=off;
CREATE TABLE test_name (
	i name
);
INSERT INTO test_name VALUES ('a'),('ab'),('abc'),('abb'),('axy'),('xyz');
CREATE INDEX idx_name ON test_name USING gin (i);
SELECT * FROM test_name WHERE i<'abc' ORDER BY i;
  i  
-----
 a
 ab
 abb
(3 rows)

SELECT * FROM test_name WHERE i<='abc' ORDER BY i;
  i  
-----
 a
 ab
 abb
 abc
(4 rows)

SELECT * FROM test_name WHERE i='abc' ORDER BY i;
  i  
-----
 abc
(1 row)

SELECT * FROM test_name WHERE i>='abc' ORDER BY i;
  i  
-----
 abc
 axy
 xyz
(3 rows)

SELECT * FROM test_name WHERE i>'abc' ORDER BY i;
  i  
-----
 axy
 xyz
(2 rows)

EXPLAIN (COSTS OFF) SELECT * FROM test_name WHERE i<'abc' ORDER BY i;
                 QUERY PLAN                  
---------------------------------------------
 Sort
   Sort Key: i
   ->  Bitmap Heap Scan on test_name
         Recheck Cond: (i < 'abc'::name)
         ->  Bitmap Index Scan on idx_name
               Index Cond: (i < 'abc'::name)
(6 rows)

EXPLAIN (COSTS OFF) SELECT * FROM test_name WHERE i<='abc' ORDER BY i;
                  QUERY PLAN                  
----------------------------------------------
 Sort
   Sort Key: i
   ->  Bitmap Heap Scan on test_name
         Recheck Cond: (i <= 'abc'::name)
         ->  Bitmap Index Scan on idx_name
               Index Cond: (i <= 'abc'::name)
(6 rows)

EXPLAIN (COSTS OFF) SELECT * FROM test_name WHERE i='abc' ORDER BY i;
              QUERY PLAN               
---------------------------------------
 Bitmap Heap Scan on test_name
   Recheck Cond: (i = 'abc'::name)
   ->  Bitmap Index Scan on idx_name
         Index Cond: (i = 'abc'::name)
(4 rows)

EXPLAIN (COSTS OFF) SELECT * FROM test_name WHERE i>='abc' ORDER BY i;
                  QUERY PLAN                  
----------------------------------------------
 Sort
   Sort Key: i
   ->  Bitmap Heap Scan on test_name
         Recheck Cond: (i >= 'abc'::name)
         ->  Bitmap Index Scan on idx_name
               Index Cond: (i >= 'abc'::name)
(6 rows)

EXPLAIN (COSTS OFF) SELECT * FROM test_name WHERE i>'abc' ORDER BY i;
                 QUERY PLAN                  
---------------------------------------------
 Sort
   Sort Key: i
   ->  Bitmap Heap Scan on test_name
         Recheck Cond: (i > 'abc'::name)
         ->  Bitmap Index Scan on idx_name
               Index Cond: (i > 'abc'::name)
(6 rows)

